src/main/java/com/github/princesslana/eriscasper/gateway/Gateway.java

```diff
@@ -156,15 +155,23 @@ private Completable heartbeat(RxWebSocket ws, Payload hello) {
   }
 
   private Completable resume(RxWebSocket ws, BotToken token) {
-    Preconditions.checkState(sessionId.isPresent(), "Can not resume without a session id");
-    Preconditions.checkState(
-        lastSeenSequenceNumber.isPresent(), "Can not resume without a sequence number");
-
     return Single.just(
-            ImmutableResume.builder()
-                .token(token)
-                .sessionId(sessionId.get())
-                .seq(lastSeenSequenceNumber.get())
+            ImmutableResumePayload.builder()
+                .token(token.unwrap())
+                .sessionId(
+                    sessionId
+                        .map(SessionId::unwrap)
+                        .orElseThrow(
+                            () ->
+                                new IllegalStateException("Can not resume without a session id.")))
+                .seq(
+                    lastSeenSequenceNumber
+                        .map(SequenceNumber::unwrap)
+                        .map(Integer::longValue)
```

Comment:
If this is meant to be a long, then SequenceNumber should probably wrap a long instead of an integer

Reply:
Agreed, I'll look through and change them as needed.